### DISCUSSION TITLE
Strange comment in generated CSS

### DISCUSSION THREAD

**Discussion by Hlusta** [tailwindlabs/tailwindcss#9122] [Help]
Hi,
my generated production CSS has some strange CSS comments:
`--tw-ring-inset: var(--tw-empty, /*!*/ /*!*/);`
see the picture for broader context:

<img width="958" alt="tw-comment" src="https://user-images.githubusercontent.com/2931374/185127853-2784e1ff-cefe-45b7-95a8-145acd628b8d.png">

My CMS unfortunately can't minificate this properly and the code above this series of comments is just avoided completely. I tried to delete all of these comments and voilà – it worked like a charm.

The question is: can I avoid these comments somehow so I do not have to delete them manually? Is it a feature and what is the purpose of it? :)

I did not spot this anywhere except this "ring-inset"...

Thank you!
Tomas

**Comment by ichenglin** [comment]


**Reply by Hlusta** [reply]
Maybe it has gone too far, but `/*!*/ /*!*/` is not a CSS comment? I know native css variables.

**Reply by tombien** [reply]
In my opinion, this
`--tw-ring-inset: var(--tw-empty, /*!*/ /*!*/);`
is an error, I think it is equal to
`--tw-ring-inset: var(--tw-empty, );`
so there is a missing attribute after comma

**Reply by RobinMalfait** [reply]
These are in fact 2 comments `/*!*/` is a comment, the `!` is typically used in comments for licenses with the `!` as a marker for "do not delete me when minifying". 

The problem here is that `--tw-empty` should actually be a space like ` ` and therefore `var(--tw-empty, )` is valid (a space), not to be confused with `var(--tw-empty, " ")` because that would be the literal `" "` which is not a space. The next problem is that a lot of minifiers think that `var(--tw-empty, )` is invalid and removed the space, but this makes a bunch of css rules invalid.

To trick certain minifiers, we did a bunch of trial and error and in fact `var(--tw-empty, /*!*/ /*!*/)` is a way to trick minifiers in leaving those comments alone resulting in a space again.

While this looks weird:
- This is intentional
- Without it, a lot of minifiers would actually break certain rules.

Hope this helps!

**Reply by Hlusta** [reply]
Thank you @RobinMalfait  for your explanation. Unfortunately OctoberCMS 2.x minifier throws all CSS above this away and starts with:
`);`
Any other suggestion is welcome!
